src/test/java/uk/gov/hmcts/cmc/claimstore/events/ccd/CCDCaseHandlerTest.java

```diff
@@ -74,4 +85,20 @@ public void saveClaimFailsWithFeignExceptionShouldTriggerAppInsight() {
             .trackEvent(CCD_ASYNC_FAILURE, REFERENCE_NUMBER, sampleClaim.getReferenceNumber());
 
     }
+
+    @Test(expected = FeignException.class)
+    public void saveLegalRepClaimFailsWithFeignExceptionShouldTriggerAppInsight() {
+        Claim sampleClaim = SampleClaim.getDefaultForLegal();
+
+        when(ccdCaseRepository.saveLegalRepClaim(user, sampleClaim))
+            .thenThrow(FeignException.class);
+
+        CCDClaimIssuedEvent claimIssuedEvent = new CCDClaimIssuedEvent(sampleClaim, user);
+        caseHandler.saveLegalRepClaim(claimIssuedEvent);
+
+        verifyZeroInteractions(ccdCaseRepository);
+        verify(appInsights, times(1))
+            .trackEvent(CCD_ASYNC_FAILURE, REFERENCE_NUMBER, sampleClaim.getReferenceNumber());
```

Comment:
These verifications will never be tested as the method will exit when the exception is thrown. Try using assertj's `assertThrownBy`:
```
assertThatThrownBy(() -> caseHandler.saveClaim(claimIssuedEvent)).isInstanceOf(FeignException.class);
```

Reply:
done